src/NHibernate/Async/Multi/QueryBatch.cs

```diff
@@ -139,33 +139,29 @@ protected async Task ExecuteBatchedAsync(CancellationToken cancellationToken)
 			}
 
 			var rowCount = 0;
+			CacheBatcher cacheBatcher = null;
 			try
 			{
 				if (resultSetsCommand.HasQueries)
 				{
+					cacheBatcher = new CacheBatcher(Session);
 					using (var reader = await (resultSetsCommand.GetReaderAsync(Timeout, cancellationToken)).ConfigureAwait(false))
 					{
-						var cacheBatcher = new CacheBatcher(Session);
 						foreach (var query in _queries)
 						{
 							if (query.CachingInformation != null)
 							{
-								foreach (var cachingInfo in query.CachingInformation.Where(ci => ci.IsCacheable))
```

Comment:
`cacheBatcher` is also needed when query is not cacheable but loaded entities are cacheable - so it's better to always set it.